src/sentry/sentryadapter.h

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef SENTRYADAPTER_H
+#define SENTRYADAPTER_H
+
+#include <QObject>
+#include <QApplication>
+
+#include <sentry.h>
+
+class SentryAdapter final : public QObject {
+  Q_OBJECT
+  Q_DISABLE_COPY_MOVE(SentryAdapter)
+
+ public:
+  ~SentryAdapter();
+  static SentryAdapter* instance();
+
+  // Inits Sentry.
+  // In case Telemetry is disabled this is a no-op.
+  void init();
+
+  // Reports an Error - can attach extra data and a stack trace if needed.
+  void report(const QString& category, const QString& message,
+              bool attachStackTrace = false);
+
+  // Called when a Line is added, will add this as a breadcrumb
+  Q_SLOT void onLoglineAdded(const QByteArray& line);
+  // Called before shutdown, will flush pending events.
+  Q_SLOT void onBeforeShutdown();
+
+  // Only define Sentry related callbacks if we have the typedefs :)
+  // Called before Sentry will send a crash report
+  static sentry_value_t onCrash(const sentry_ucontext_t* uctx,
+                                sentry_value_t event, void* closure);
+
+  // Will be used if NONE_TRANSPORT is enabled
+  // in that case this will be called if the client has a
+  // report to send to Sentry.
+  static void transportEnvelope(sentry_envelope_t* envelope, void* state);
```

Comment:
Love the comment adding on this file. In mozilla-central docstrings are added to the header files too, right? I've been adding them to the .cpp file, maybe we should reach an agreement about this, happy to swich from .cpp to .h. On another note, I believe ideally we should use doxygen format comments, just because if we ever decide to auto generate docs based on these doc-comments it will be easier that way.

Reply:
mc has most in the header; Which i kind of like because I can easily can get a glimpse of what everything in the class is happening. :D 
Totally with you about the standard comment, I already forgot that we agreed on a standard format :D

Reply:
LOL, well me and you agreed. We should probably broadcast that. I agree with having it in the header. I'll do that from now on and also change the stuff that is not in the header!

Reply:
File looks beautiful now with all the comments :100: